app/models/search_result.rb

```diff
@@ -23,4 +23,40 @@ def initialize(_params)
           results_count: [repositories_count,
                           organizational_units_count].sum)
   end
+
+  # We can refactor the whole class when we drop the JSON API, for now, this
+  # returns the search result in a format that graphql can work with without
+  # influencing the JSON API controller
+  # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def to_struct(params)
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
+    result = OpenStruct.new
+    repos = repositories.map do |repo|
+      OpenStruct.new(entry: RepositoryCompound.wrap(repo), ranking: 1.0)
+    end
+    org_units = organizational_units.map do |org_unit|
+      OpenStruct.new(entry: org_unit, ranking: 1.0)
+    end
+
+    result.entries = []
+    if !params[:categories] ||
+       params[:categories].empty? ||
+       params[:categories].include?('repositories')
```

Comment:
I think this can be made simpler:
```ruby
if params[:categories].blank? || params[:categories].include?('repositories')
```
See http://api.rubyonrails.org/classes/Object.html#method-i-blank-3F